service/controller/v1/resource/app/resource.go

```diff
@@ -66,23 +66,17 @@ func (r *Resource) Name() string {
 	return Name
 }
 
-func containsAppCRs(list []*applicationv1alpha1.App, item *applicationv1alpha1.App) bool {
-	_, err := getAppCRByName(list, item.Name)
-	if err != nil {
-		return false
-	}
-
-	return true
-}
-
-func getAppCRByName(list []*applicationv1alpha1.App, name string) (*applicationv1alpha1.App, error) {
+func getAppCR(list []*applicationv1alpha1.App, namespace, name string) (*applicationv1alpha1.App, bool) {
 	for _, l := range list {
-		if l.Name == name {
-			return l, nil
+		b := true
+		b := b && l.Namespace == namespace
+		b := b && l.Name == name
+		if b {
```

Comment:
`if l.Namespace == namespace && l.Name == name {` would have been fine also.

Reply:
Yeah I'm just getting ready for the future that will never happen :D